### DISCUSSION TITLE
If Documentation

### DISCUSSION THREAD

**Discussion by TofuBug** [JanDeDobbeleer/oh-my-posh#2286] [Troubleshoot] [answered]
Hello all,

I hope I'm just going to be embarrassed when someone points out what I've just overlooked, but I'm finding it impossible to find a  comprehensive documentation on the options and limitations of the if statements in templates

Is it just limited to implicit equality, lt, gt, and, or etc

is there support for things e.g. like, in, contains, etc and if so where is the documentation for the syntax of these statements

I'd be perfectly happy with a Backus-Nuar document on the If Statement in a template string but obviously a more human readable document would be better for all involved.

Thanks in advance

**Comment by JanDeDobbeleer** [comment, marked as answer]
@TofuBug incoming in #2287

**Reply by TofuBug** [reply]
LOL well that was a little uncanny getting a commit in less than 10 hours with exactly what I was randomly looking for.

Question though is there any plan to include logic for other boolean logic e.g. Like/Regex(Match), Less Than or Equal, Greater Than or Equal in the future?

**Reply by JanDeDobbeleer** [reply]
> is there any plan to include logic for other boolean logic e.g. Like/Regex(Match), Less Than or Equal, Greater Than or Equal in the future?

@TofuBug this is already present, as they're part of the standard stack. See the last entry [here](https://ohmyposh.dev/docs/configuration/templates#template-logic):

> This lt comparison function evaluates to true since 3 is less than 4 (other boolean operators: eq, ne, lt, le, gt, ge).

As for regex and match, that not available. But maybe the sprig [string functions](https://masterminds.github.io/sprig/strings.html) already do a lot?
